File: Astromaximum/Classes/AmaxDateSelectController.m
//
//  AmaxDateSelectController.m
//  Astromaximum
//
//  Created by admin on 04.12.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

#import "AmaxDateSelectController.h"
#import "AmaxDataProvider.h"

@implementation AmaxDateSelectController
@synthesize datePicker;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone   target:self action:@selector(didSelectDate:)];
    }
    return self;
}

- (void)didReceiveMemoryWarning
{
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc that aren't in use.
}

#pragma mark - View lifecycle

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

- (void)viewDidUnload
{
    [self setDatePicker:nil];
    [super viewDidUnload];
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    // Return YES for supported orientations
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}

- (IBAction)didSelectDate:(id)sender
{
    [[AmaxDataProvider sharedInstance]setDateFrom:[datePicker date]];
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: Astromaximum/Classes/AmaxDateSelectController.swift
//
//  AmaxDateSelectController.m
//  Astromaximum
//
//  Created by admin on 04.12.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

import UIKit

class AmaxDateSelectController : UIViewController {

    @IBOutlet weak var datePicker: UIDatePicker?

    override init(nibName: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName:nibName, bundle:nibBundleOrNil)
        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem:.done, target:self, action:#selector(self.didSelectDate(sender:)))
        title = NSLocalizedString("pick_date", comment: "Pick date")
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func didReceiveMemoryWarning() {
        // Releases the view if it doesn't have a superview.
        super.didReceiveMemoryWarning()

        // Release any cached data, images, etc that aren't in use.
    }

    // MARK: - View lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        datePicker?.minimumDate = Date(timeIntervalSince1970: TimeInterval(AmaxDataProvider.sharedInstance.mStartJD))
        datePicker?.maximumDate = Date(timeIntervalSince1970: TimeInterval(AmaxDataProvider.sharedInstance.mFinalJD))
        let date = AmaxDataProvider.sharedInstance.currentDate()
        //print(date)
        datePicker!.setDate(date, animated: false)
    }
    
    @IBAction func didSelectDate(sender:AnyObject!) {
        AmaxDataProvider.sharedInstance.setDate(from: datePicker!.date)
        self.navigationController?.popViewController(animated: true)
    }
}
